python/common/instrument.py

```diff
@@ -39,26 +50,28 @@ def __init__(self, frame, filename=None):
             if hasattr(self.frame, '_instrument_headers'):
                 headers.extend(self.frame._instrument_headers)
 
-            self.fieldnames = headers
-            writer = csv.DictWriter(csvfile, delimiter=',', lineterminator='\n', fieldnames=self.fieldnames)
+            self.fieldnames = list(set(headers))
```

Comment:
Why list(set()) why not just list()?
It just seems to be scrambling the order?
Is that what you want to do? If yes, comment please.